### PR TITLE
add tagged release workflow using github actions

### PR DESCRIPTION
**Pull request by jvandew**
### Description
This PR adds a new git tag based release workflow, as outlined in #1450, mirroring the setup that exists in other Bazel rulesets. Pushing a versioned git tag of the form `v1.2.3` will trigger a Github action to generate and document a release archive.

The setup here is largely identical to the release actions in [rules_python](https://github.com/bazelbuild/rules_python/blob/00545742ad2450863aeb82353d4275a1e5ed3f24/.github/workflows/release.yml) and [rules_js](https://github.com/aspect-build/rules_js/blob/fb1f84aca12627308422097e433fb2eb52b0cf23/.github/workflows/release.yml), though for simplicity it omits the test step as being redundant to the existing CI setup.

There are some sample runs in my fork [here](https://github.com/jvandew/rules_scala/actions/runs/3558745957), which generate releases such as [this](https://github.com/jvandew/rules_scala/releases/tag/v1.2.12).

### Motivation
Addresses #1450, originally motivated by a [slack request](https://bazelbuild.slack.com/archives/CDCKJ2KFZ/p1668601467490499).

### CODE REVIEW

**Comment by liucijus** [comment]
Thanks @jvandew!
What would we potentially miss if we skip a testing step? Rules Scala test runs are very heavy, maybe we can rely on master builds? I don't have my opinion formed yet, would like to hear any pros/cons. 
If we have to test, then we need to run `./test_rules_scala.sh`

**Comment by jvandew** [comment]
I'm not familiar with the existing build setup in rules_scala (looks like the master build you're referring to is scheduled daily on [Buildkite](https://buildkite.com/bazel/rules-scala-scala)?), but relying on the master builds seems reasonable to me... The main risk I think would be that two PRs individually pass CI but conflict when merged together, and a release is cut before the scheduled master build runs to detect the break, but that is probably a rare occurrence and it is probably easy enough to just wait for a green build on master before releasing.

The flip side I guess is that Github Actions appears to have fairly generous [resource limits](https://docs.github.com/en/actions/learn-github-actions/usage-limits-billing-and-administration) for public repos, so there probably isn't an actual cost to running a heavy test suite beyond the time it takes to execute.

**Comment by liucijus** [comment]
@jvandew maybe let's continue without testing step, if we see that testing is needed, we can re-add it later. Running without testing would simplify release a lot.

**Comment by jvandew** [comment]
> @jvandew maybe let's continue without testing step, if we see that testing is needed, we can re-add it later. Running without testing would simplify release a lot.

updated! the release is [quite snappy](https://github.com/jvandew/rules_scala/actions/runs/3635653305) now

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Prepare workspace snippet
+        run: .github/workflows/workspace_snippet.sh ${{ env.GITHUB_REF_NAME }} > release_notes.txt
```
Comment (simuons) [inline .github/workflows/release.yml:18]: Thanks @jvandew, this is really cool. Just a small question: does `${{ env.GITHUB_REF_NAME }}` needs to be passed here? Cannot find where it's taken in the script, looks like it's read from env.

Comment (jvandew) [reply .github/workflows/release.yml:18]: aha you appear to be correct, nice catch!

**Review by liucijus — APPROVED** [review]
Thanks, @jvandew!

**Pull request merged**